File: frontend/src/service/todoService.js
import axiosClient from '../api/axiosClient';


const todoService = {
    getTodosForCurrentWeek : async (week,status,priority,title) => {
        const response = await axiosClient.get('todos/week', { params: { week, status, priority, title } });
        return response.data;
    },
    createTodo: async (todoData) => {
        const response = await axiosClient.post('todos', todoData);
        return response.data;
    },
    editTodo: async (todoId, todoData) => {
        const response = await axiosClient.put(`todos/${todoId}`, todoData);
        return response.data;
    },
    deleteTodo: async (todoId) => {
        const response = await axiosClient.delete(`todos/${todoId}`);
        return response.data;
    },
    getTodoById : async (todoId) => {
        const response = await axiosClient.get(`todos/${todoId}`);
        return response.data;
    }
}

export default todoService;

File: backend/src/main/java/iuh/fit/todo_app/service/ToDoService.java
package iuh.fit.todo_app.service;

import iuh.fit.todo_app.dto.TodoRequest;
import iuh.fit.todo_app.entity.Todo;
import iuh.fit.todo_app.enums.Priority;
import iuh.fit.todo_app.enums.Status;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

public interface ToDoService {

    Map<LocalDate, Map<String, List<Todo>>> getTodosByWeekAndFilters(Status status, Priority priority, String title, int weekOffset);

    Todo createTodo(TodoRequest todoRequest);

    Todo editTodo(Integer id, TodoRequest todoRequest);

    void deleteTodo(Integer id);

    Todo getTodoById(Integer id);
}
